lapce-ui/src/editor/tab_header_content.rs

```diff
@@ -248,6 +249,79 @@ impl LapceEditorTabHeaderContent {
             ));
         }
     }
+
+    fn get_effective_path(
+        &mut self,
+        workspace_path: Option<PathBuf>,
+        file_path: &PathBuf,
+    ) -> String {
+        let workspace_path = match workspace_path {
+            Some(p) => p,
+            None => {
+                let mut file_path_str = match file_path.to_str() {
+                    Some(p) => p.to_string(),
+                    None => return "".to_string(),
+                };
+
+                file_path_str.truncate(20);
```

Comment:
This has potential issue if the name contains special characters, since 20 might lie in the middle of a multibyte character which would panic.  
I add some issue in finding if there was a better solution than truncate for this task, unfortunately.  
```rust
let file_path_str = file_path_str.char_indices()
        .filter(|(i, _)| *i < 20)
        .map(|(_, ch)| ch)
        .collect::<String>();
```
Ugly, and I feel like there should be a better way, but I believe it should work.

Reply:
It looks like it works like a charm yeah! I'll add to test for unicode-based names to avoid regressions.